server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java

```diff
@@ -186,7 +187,7 @@ public void run() {
         } catch (IOException | RuntimeException ex) {
           fail(work, ex, "synching");
         }
-        long duration = System.currentTimeMillis() - start;
+        long duration = NANOSECONDS.toMillis(System.nanoTime() - start);
         if (duration > slowFlushMillis) {
```

Comment:
I think it would be best to convert `slowFlushMillis` to nanos and then keep `start` and `duration` as nanos too. That way you are getting the benefits of using nanos without having to loose precision by converting back to millis.